### DISCUSSION TITLE
[question] Reverse proxy in docker for DoH

### DISCUSSION THREAD

**Discussion by mercurykd** [AdguardTeam/AdGuardHome#5355] [Q&A]
closed

**Comment by SickProdigy** [comment]
I am wondering something similar. Adguard behind nginx proxy (swag) server. Now I am thinking nginx server should be behind adguard. You can map the ports to different area within docker (and your router if you wanted) 444(host):443(container)
Nginx needs the host 443 to display websites properly, adguard can just be mapped in anywhere I suppose.

Thoughts:
Router External (internet) 443 >> Internal 444 adguard >> Have adguard uplink through nginx reverse proxy?

Just kind of stuck on how it should be done. Still learning docker networking.

**Comment by sh3bang** [comment]
Hello, 

same problem, want to setup Adguards DoH behind Nginx, but no success:

what is the solution to this topic?
https://github.com/AdguardTeam/AdGuardHome/issues/5334

SOLVED: Config is ok and resolving by DoH also, but not with this tool, because "CORS" will fail during Testing:
https://dnsleaktest.org/dns-over-https

**Reply by SickProdigy** [reply]
My setup went broke not too long ago. Having issues with port 3000 showing behind nginx. Can I see your nginx config for adguard?

**Comment by sh3bang** [comment]
this topic is related to DoH behind nginx, but this should help:

https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#nginx
